### ISSUE TITLE
Secure thread implementation.

### ISSUE THREAD

**Issue by adriancarayol** [proofit404/asgi_rabbitmq#12]
@proofit404 - have you considered using amqpstorm instead of pika?
https://github.com/eandersson/amqpstorm

**Comment by proofit404** [comment]
Hi! Thanks for your interest in my project!

I'm not familiar with amqpstorm library. At least we need to answer the same questions as we do in #9.

Do you have any troubles with existing `asgi_rabbitmq` implementation? I think I  solve threading issues on my side.

**Comment by adriancarayol** [comment]
Hi! asgi_rabbitmq + rabbitmq sometimes causes errors, and the whole application crashes.

Seems somewhat unstable in production

**Comment by proofit404** [comment]
@adriancarayol thanks for example project I can test possible solution. If I understand correctly moving from pika 0.10 to 0.11 should solve related framing issue.

**Comment by kfrendrich** [comment]
Hi.

I have the same problem that @adriancarayol mentioned.
Unfortunately the pika upgrade/downgrade can't solve it.

```
Error processing message with consumer channels.handler.ViewConsumer:
Traceback (most recent call last):
File "/usr/local/lib/python3.5/dist-packages/channels/worker.py", line 119, in run
consumer(message, **kwargs)
File "/usr/local/lib/python3.5/dist-packages/channels/handler.py", line 360, in __call__
message.reply_channel.send(reply_message, immediately=True)
File "/usr/local/lib/python3.5/dist-packages/channels/channel.py", line 44, in send
self.channel_layer.send(self.name, content)
File "/usr/local/lib/python3.5/dist-packages/asgi_rabbitmq/core.py", line 651, in send
return future.result()
File "/usr/lib/python3.5/concurrent/futures/_base.py", line 405, in result
return self.__get_result()
File "/usr/lib/python3.5/concurrent/futures/_base.py", line 357, in __get_result
raise self._exception
pika.exceptions.ConnectionClosed

Traceback (most recent call last):
File "manage.py", line 12, in <module>
execute_from_command_line(sys.argv)
File "/usr/local/lib/python3.5/dist-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
utility.execute()
File "/usr/local/lib/python3.5/dist-packages/django/core/management/__init__.py", line 356, in execute
self.fetch_command(subcommand).run_from_argv(self.argv)
File "/usr/local/lib/python3.5/dist-packages/django/core/management/base.py", line 283, in run_from_argv
self.execute(*args, **cmd_options)
File "/usr/local/lib/python3.5/dist-packages/django/core/management/base.py", line 330, in execute
output = self.handle(*args, **options)
File "/usr/local/lib/python3.5/dist-packages/channels/management/commands/runworker.py", line 83, in handle
worker.run()
File "/usr/local/lib/python3.5/dist-packages/channels/worker.py", line 87, in run
channel, content = self.channel_layer.receive_many(channels, block=True)
File "/usr/local/lib/python3.5/dist-packages/asgiref/base_layer.py", line 43, in receive_many
return self.receive(channels, block)
File "/usr/local/lib/python3.5/dist-packages/asgi_rabbitmq/core.py", line 658, in receive
future = self.thread.schedule(RECEIVE, channels, block)
File "/usr/local/lib/python3.5/dist-packages/asgi_rabbitmq/core.py", line 593, in schedule
return self.connection.schedule(f, *args, **kwargs)
File "/usr/local/lib/python3.5/dist-packages/asgi_rabbitmq/core.py", line 558, in schedule
raise ConnectionClosed
pika.exceptions.ConnectionClosed
```

RabbitMQ logs:
```
operation none caused a connection exception frame_error: "type 3, first 16 octets = <<\"3209,1385255009,\">>: {invalid_frame_end_marker,\n 48}"
```

**Comment by kfrendrich** [comment]
Hi.
I have the same problem with pika that @adriancarayol mentioned.
Unfortunately the pika upgrade/downgrade can't solve it.

```
Error processing message with consumer channels.handler.ViewConsumer:
Traceback (most recent call last):
File "/usr/local/lib/python3.5/dist-packages/channels/worker.py", line 119, in run
consumer(message, **kwargs)
File "/usr/local/lib/python3.5/dist-packages/channels/handler.py", line 360, in __call__
message.reply_channel.send(reply_message, immediately=True)
File "/usr/local/lib/python3.5/dist-packages/channels/channel.py", line 44, in send
self.channel_layer.send(self.name, content)
File "/usr/local/lib/python3.5/dist-packages/asgi_rabbitmq/core.py", line 651, in send
return future.result()
File "/usr/lib/python3.5/concurrent/futures/_base.py", line 405, in result
return self.__get_result()
File "/usr/lib/python3.5/concurrent/futures/_base.py", line 357, in __get_result
raise self._exception
pika.exceptions.ConnectionClosed
Traceback (most recent call last):
File "manage.py", line 12, in <module>
execute_from_command_line(sys.argv)
File "/usr/local/lib/python3.5/dist-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
utility.execute()
File "/usr/local/lib/python3.5/dist-packages/django/core/management/__init__.py", line 356, in execute
self.fetch_command(subcommand).run_from_argv(self.argv)
File "/usr/local/lib/python3.5/dist-packages/django/core/management/base.py", line 283, in run_from_argv
self.execute(*args, **cmd_options)
File "/usr/local/lib/python3.5/dist-packages/django/core/management/base.py", line 330, in execute
output = self.handle(*args, **options)
File "/usr/local/lib/python3.5/dist-packages/channels/management/commands/runworker.py", line 83, in handle
worker.run()
File "/usr/local/lib/python3.5/dist-packages/channels/worker.py", line 87, in run
channel, content = self.channel_layer.receive_many(channels, block=True)
File "/usr/local/lib/python3.5/dist-packages/asgiref/base_layer.py", line 43, in receive_many
return self.receive(channels, block)
File "/usr/local/lib/python3.5/dist-packages/asgi_rabbitmq/core.py", line 658, in receive
future = self.thread.schedule(RECEIVE, channels, block)
File "/usr/local/lib/python3.5/dist-packages/asgi_rabbitmq/core.py", line 593, in schedule
return self.connection.schedule(f, *args, **kwargs)
File "/usr/local/lib/python3.5/dist-packages/asgi_rabbitmq/core.py", line 558, in schedule
raise ConnectionClosed
pika.exceptions.ConnectionClosed

```

rabitmq logs:

```
operation none caused a connection exception frame_error: "type 3, first 16 octets = <<\"3209,1385255009,\">>: {invalid_frame_end_marker,\n 48}"

```

**Comment by proofit404** [comment]
Related issue was solved in 9d576592d9af784d478af2edaecd1fdf9f2e8a94

Ref: https://github.com/proofit404/asgi_rabbitmq/pull/14

**Comment by proofit404** [comment]
Fixed in `0.5.4`. The release is already on PyPI.
